### PR TITLE
[Modules] Updated Resources/DeploymentScripts to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Resources/DeploymentScripts to new dependency approach

| Pipeline |
| - |
| [Resources: DeploymentScripts](https://github.com/Azure/ResourceModules/actions/workflows/ms.resources.deploymentscripts.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,56 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    azPowerShellVersion: '3.0'
```
Comment (eriqua) [inline modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep:44]:
```suggestion
    azPowerShellVersion: '8.0'
```

File: modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    cleanupPreference: 'Always'
+    kind: 'AzurePowerShell'
+    lock: 'CanNotDelete'
+    retentionInterval: 'P1D'
+    runOnce: false
+    scriptContent: 'Write-Host \'Running PowerShell from template\''
```
Comment (eriqua) [inline modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep:50]: Boooriiiiing, but OK

File: modules/Microsoft.Resources/deploymentScripts/.test/cli/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    azCliVersion: '2.15.0'
```
Comment (eriqua) [inline modules/Microsoft.Resources/deploymentScripts/.test/cli/deploy.test.bicep:44]:
```suggestion
    azCliVersion: '2.41.0'
```

File: modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep (hunk 3 above)
Comment (MrMCake) [reply modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep:50]: Will update

Comment (MrMCake) [reply modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep:50]: done

Comment (eriqua) [reply modules/Microsoft.Resources/deploymentScripts/.test/ps/deploy.test.bicep]: Appreciate the effort 🍰

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**